plugin/common/common.cuh

```diff
@@ -322,66 +322,54 @@ __device__ inline void scaledSoftmaxSmall(
     }
 }
 
-template <typename T, unsigned TPB>
-__device__ inline void scaledSoftmax(
-    const int ld, const int lastValid, const float rsqrtHeadSize, const T* input, T* output)
-{
-
-    using BlockReduce = cub::BlockReduce<float, TPB>;
-    __shared__ typename BlockReduce::TempStorage tmpStorage;
-
-    __shared__ float rZ;
-    __shared__ float fMax;
-
-    const int offset = (blockIdx.y * gridDim.x + blockIdx.x) * ld;
-
-    const float w(rsqrtHeadSize);
-    cub::Sum sum;
-    float threadData(-FLT_MAX);
-
-    if (lastValid >= blockDim.x)
```

Comment:
Could we keep this check here in case the kernel is used in scenario `TPB > ld`? thanks!

Reply:
Yeah, you can keep it. Actually, this check confused me a lot. From my point of view, with this check, the initial value of `threadData` will be 0 in the scenario `TPB <= lastValid` but -FLT_MAX in the scenario `TPB > lastValid`. However, `threadData` will definitely be updated by `threadData = max(static_cast<float>(input[idx]), threadData);` in the next loop, so does this check is necessary for threads in the block? Or could we directly initialize the `threadData` by 0 instead of `-FLT_MAX`?

Reply:
the `threadData = max(static_cast<float>(input[idx])` is only called on condition `i < lastValid`, if `TPB <= lastValid`, we will accumulate the `-FLT_MAX` in next block reduced sum.
If we initialize the threadData to 0, then the `threadData = max(static_cast<float>(input[idx])` is invalid if the input is negative.

Reply:
From my point of view, maybe we just try to get the fmax here for `BlockReduceMax`? I'll set `threadData` as 0 before `BlockReduceSum`.

Reply:
I mean line 347